client/src/modules/dashboard/Events/components/EventForm.tsx

```diff
@@ -125,24 +132,52 @@ const EventForm: React.FC<EventFormProps> = (props) => {
           Invite only
         </Checkbox>
 
-        {loadingVenues ? (
-          <h1>Loading venues...</h1>
-        ) : errorVenues || !dataVenues ? (
-          <h1>Error loading venues</h1>
-        ) : (
-          <FormControl>
-            <FormLabel>Venue</FormLabel>
-            <Select {...register('venue_id')}>
-              {dataVenues.venues.map((v) => (
-                <option key={v.id} value={v.id}>
-                  {v.name}
-                </option>
+        <FormControl isRequired>
+          <FormLabel>Venue Type</FormLabel>
+          <RadioGroup defaultValue={venueType}>
+            <HStack>
+              {venueTypes.map((venueType) => (
+                <Radio
+                  key={venueType.value}
+                  value={venueType.value}
+                  {...register('venue_type')}
+                >
+                  {venueType.name}
+                </Radio>
               ))}
+            </HStack>
+          </RadioGroup>
 
-              <option>None</option>
-            </Select>
-          </FormControl>
-        )}
+          {loadingVenues ? (
+            <h1>Loading venues...</h1>
+          ) : errorVenues || !dataVenues ? (
+            <h1>Error loading venues</h1>
+          ) : (
+            getValues('venue_type') !== VenueType.Online && (
+              <FormControl isRequired>
+                <FormLabel>Venue</FormLabel>
+                <Select {...register('venue_id')}>
+                  {dataVenues.venues.map((v) => (
+                    <option key={v.id} value={v.id}>
+                      {v.name}
+                    </option>
+                  ))}
+                </Select>
+              </FormControl>
+            )
+          )}
+
+          {getValues('venue_type') !== VenueType.Physical && (
```

Comment:
This gave me idea. What do you think about adding also `FieldResolver` for both `isOnline` and `isPhysical` fields in the event resolver? This way client could query those fields, and use the provided by server values, instead of needing to deal with that again.

Reply:
I've found one file in client that would need helper function anyway, so I've for now left idea of custom `FieldResolver`s and added functions.

Reply:
Oh, that's a shame - I liked that idea.  Which file caused the problem?

Reply:
> Unless helpers usage would be limited only to that file, and in other places in client, use queried values from FieldResolvers? Wouldn't be that too strange?

I think that would be a little odd, yeah.  It wouldn't be clear (without this context) why sometimes helpers are used and sometimes the fields are accessed directly.

Fundamentally, there are exactly three choices and the client needs to know them all to create the radio button.  Oh well, helpers will do!